src/main/scheduler.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+typedef enum {
+    TASK_PRIORITY_IDLE = 0,     // Disables dynamic scheduling, task is executed only if no other task is active this cycle
+    TASK_PRIORITY_LOW = 1,
+    TASK_PRIORITY_MEDIUM = 3,
+    TASK_PRIORITY_HIGH = 5,
+    TASK_PRIORITY_REALTIME = 6,
+    TASK_PRIORITY_MAX = 255
+} cfTaskPriority_e;
+
+typedef struct {
+    const char * taskName;
+    bool         isEnabled;
+    uint32_t     desiredPeriod;
+    uint8_t      staticPriority;
+    uint32_t     maxExecutionTime;
+    uint32_t     totalExecutionTime;
+} cfTaskInfo_t;
+
+typedef enum {
+    /* Actual tasks */
+    TASK_SYSTEM = 0,
+    TASK_GYROPID,
```

Comment:
I've tried this, but it gives too much overhead - 20us scheduling delay for each of the tasks. Also, the scheduler is limited to one realtime task, otherwise jitter gets introduced.